### PR TITLE
Register custom datafiles maker in pytest

### LINKED ISSUE #8: PytestUnknownMarkWarning with pytest 4.3.0
With pytest 4.3.0 I am constantly seeing the following warning when executing tests using datafiles:
```
.tox/test-py37/lib/python3.7/site-packages/_pytest/mark/structures.py:324
  xxxx/.tox/test-py37/lib/python3.7/site-packages/_pytest/mark/structures.py:324: PytestUnknownMarkWarning: Unknown pytest.mark.datafiles - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/latest/mark.html
    PytestUnknownMarkWarning,
```
The plugin works as expected though

### PR DESCRIPTION
**Pull request by dazza-codes**
Register custom makers, see
- https://docs.pytest.org/en/latest/writing_plugins.html#registering-custom-markers

Fix #8

### CODE REVIEW

File: pytest_datafiles.py [hunk 1]
```diff
@@ -5,6 +5,11 @@
 from py import path  # pylint: disable=E0611
 import pytest
 
+def pytest_configure(config):
+    config.addinivalue_line("datafile", "datafile: use tmp data files.")
```
Comment (omarkohl) [inline pytest_datafiles.py]: The build probably fails because it should be "datafiles" (plural).

Comment (darrenleeweber) [reply pytest_datafiles.py]: Thanks, done.

**Comment by darrenleeweber** [comment]
/shrug - it doesn't work as advertised, too complex for my time schedule to warrant further investigation - feel free to trash this PR and do it over again, sorry I can't follow through.

**Comment by darrenleeweber** [comment]
Try only adding one ini-line.

**Comment by darrenleeweber** [comment]
Related issue upstream at https://github.com/pytest-dev/pytest/issues/5351

File: pytest_datafiles.py [hunk 2]
```diff
@@ -5,6 +5,8 @@
 from py import path  # pylint: disable=E0611
 import pytest
 
+def pytest_configure(config):
+    config.addinivalue_line("datafiles", "datafiles(path): path to tmp data files")
```
Comment (nicoddemus) [inline pytest_datafiles.py]:
```suggestion
    config.addinivalue_line("markers", "datafiles(path): path to tmp data files")
```

As per docs: https://docs.pytest.org/en/latest/mark.html#registering-marks

Comment (darrenleeweber) [reply pytest_datafiles.py]: Right on.  Yes-Sir!  (Evoking memories of Jack Nicklaus winning the 1986 masters.)

**Comment by tsvi** [comment]
Looking forward to having this merged.
As a temporary solution I added the following to my `setup.cfg`:

    [tool:pytest]
    markers =
         datafiles(path): path to tmp data files

**Comment by dazza-codes** [comment]
Bump for merge?

**Comment by nicoddemus** [comment]
Thanks @dazza-codes!

**Comment by pvdissel** [comment]
Bump for merge?

**Comment by omarkohl** [comment]
Sorry for the long delay, I will have a look shortly

**Comment by omarkohl** [comment]
Thanks for the pull request! Since the 'datafiles' marker takes multiple parameters this code was only half correct and I fixed it in commit f95315ee2dff86ffeb17a41c203d54c1fc1405f0 instead of merging it.

**Pull request closed without merging**